cibuildwheel/linux.py

```diff
@@ -78,8 +71,23 @@ def get_python_configurations(build_selector: BuildSelector) -> List[PythonConfi
         PythonConfiguration(version='3.8', identifier='cp38-manylinux_s390x', path_str='/opt/python/cp38-cp38'),
         PythonConfiguration(version='3.9', identifier='cp39-manylinux_s390x', path_str='/opt/python/cp39-cp39'),
     ]
+
     # skip builds as required
-    return [c for c in python_configurations if matches_platform(c.identifier) and build_selector(c.identifier)]
+    target_archs = architectures
+
+    if Architecture.auto in architectures:
+        target_archs.remove(Architecture.auto)
+        native_architecture = Architecture(platform.machine())
+        target_archs.append(native_architecture)
+        # x86_64 machines can run i686 docker containers
+        if native_architecture == Architecture.x86_64:
```

Comment:
if i686 is already selected (+x86+64), it will duplicate it. I think it's harmless though